www/src/components/Toc.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const Toc = ({ data, className }) => {
+  const generateTree = (headings) => (headings?.items?.length
+    ? (
+      <ul className="pgn-doc__toc-list">
+        {headings.items.map(heading => (
+          <li key={heading.url}>
+            <a href={heading.url}>{heading.title}</a>
+            {!!heading.items && (
+            <ul className="pgn-doc__toc-list">
+              {generateTree(heading)}
+            </ul>
+            )}
+          </li>
+        ))}
+      </ul>
+    ) : 'No headings on the page');
+
+  return (
+    <div className={classNames(className, 'pgn-doc__toc')}>
+      <p className="pgn-doc__toc-header">Contents</p>
+      {generateTree(data)}
+    </div>
+  );
+};
+
+Toc.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
```

Comment:
nit: Could we use `PropTypes.shape({ ... })` to avoid the eslint error?

Reply:
added recursive `PropTypes` for this